lib/raven/event.rb

```diff
@@ -134,6 +134,22 @@ def from_message(message, options = {})
 
       private
 
+      def get_exception_class(x)
+        x.is_a?(Module) ? x : qualified_const_get(x)
+      rescue NameError # There's no way to safely ask if a constant exist for an unknown string
+        nil
+      end
+
+      # In Ruby <2.0 const_get can't lookup "SomeModule::SomeClass" in one go
+      def qualified_const_get(x)
+        parts = x.to_s.split("::").delete_if(&:empty?)
+        if parts.size < 2
+          Object.const_get(x.to_s)
```

Comment:
Hmm, not sure it's worth it, we're not completely exception-safe even with `const_defined?` (this was not the what I expecting).

```Ruby
[1] pry(main)> Object.const_defined?("String")
=> true
[2] pry(main)> Object.const_defined?("string")
NameError: wrong constant name string
from (pry):2:in `const_defined?'
```

I did a quick benchmark comparing a rescue and checking with `const_defined?` (this was before I realised `const_defined?` could raise exceptions for non-camel-case strings). When the constant is undefined the rescue-version was actually faster. This shouldn't be that performance sensitive since it's only run when exceptions are being captured, but still. :)
https://gist.github.com/jage/b0d3c397b5b46de421b537cbe41219f7#benchmark-how-to-look-up-a-class-from-string-in-ruby

Still think we should do a check before getting the constant?